packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -360,9 +360,10 @@ export function useSelectableCollection(options: AriaSelectableCollectionOptions
       let element = scrollRef.current.querySelector(`[data-key="${manager.focusedKey}"]`) as HTMLElement;
       if (element) {
         scrollIntoView(scrollRef.current, element);
+        scrollIntoViewport(element, ref.current);
```

Comment:
both and opposite signatures?

Reply:
yeaaah. The ideal (?) flow is as follows:
1. Scroll the newly focused element into view with respect to the scrollView. Limit this strictly to the first scrollParent (pre-existing behavior aka `scrollIntoView(scrollRef.current, element);`). This means the element may still be out of view for a variety of reasons (nested scroll regions, collection is extends past the viewport, something is covering the element) 
2. Scroll everything in such a way that the element is actually visible to the user (aka scrollIntoViewport(element, ref.current);`. If `preventScroll` is on, we only scroll up to the body so that we don't accidentally shift any overlays that maybe open 

The opposite signatures is a bit unfortunate, the second arg to `scrollIntoViewport` is the `containerElement` and is optional hence why it is the second arg (e.g. if we were scrolling a button into view, that case isn't actually implemented in this PR). I'll think a bit about the api here

Reply:
Just to clarify, the suggestion here is to flip `scrollIntoView`'s arguments to match `scrollIntoViewport` arg order? Or is it to make `scrollIntoView`'s signature the following -> `scrollIntoView(objectArg, element)` where objectArg is `{element: Element, scrollRef: Element}`? Or is it to modify `scrollIntoViewport`'s signature?

Reply:
make both of them into
`functionName(objectArg)` where objectArg is what you outlined
you'd need to maintain the current signature for scrollIntoView as well, which if the args aren't optional, then it's pretty easy to distinguish, one arg would be the new signature and 2 would be the old, you'd use TS overload syntax

Reply:
Hm, I see. I'm a bit hesitant to combine them into a single function since they fulfill slightly different purposes. `scrollIntoView` scrolls a element into view with respect to its scroll parent (not necessarily scrolling it visually into the viewport) and `scrollIntoViewport` brings a element into the viewport by centering its containing element (not necessarily the scroll parent or anything) in the viewport. It also uses `scrollIntoView` in the case where preventScroll is active so that function needs to exist as a utility